src/components/top-bar/_top-bar.scss

```diff
@@ -90,17 +93,24 @@
 }
 
 .player.size-sm .top-bar {
-  padding: #{$top-bar-top-bottom-gutter}px #{$gui-small-gutter}px;
-
   .left-controls {
     margin: #{$top-bar-top-bottom-gutter}px 0 #{$top-bar-top-bottom-gutter}px #{$gui-small-gutter}px;
+    &:empty {
+      margin: 0 0 #{$gui-gutter}px 0;
+    }
   }
 
   .right-controls {
     margin: #{$top-bar-top-bottom-gutter}px #{$gui-small-gutter}px #{$top-bar-top-bottom-gutter}px 0;
+    &:empty {
+      margin: 0 0 #{$gui-gutter}px 0;
+    }
   }
   .top-bar-area {
     width: calc(100% - #{2 * $gui-small-gutter}px);
     margin: #{$top-bar-top-bottom-gutter}px #{$gui-small-gutter}px;
+    &:empty {
+      margin: 0 0 #{$gui-gutter}px 0;
+    }
   }
 }
```

Comment:
This solves another bug in top bar - padding is not used anymore since we added margins to the children and also the margin css was overridden from the regular size hover margin.